action/protocol/staking/handlers.go

```diff
@@ -382,6 +387,11 @@ func (p *Protocol) handleTransferStake(ctx context.Context, act *action.Transfer
 	}
 	log.AddTopics(byteutil.Uint64ToBytesBigEndian(bucket.Index), act.VoterAddress().Bytes(), bucket.Candidate.Bytes())
 
+	if p.hu.IsPost(config.Hawaii, blkCtx.BlockHeight) && address.Equal(newOwner, bucket.Owner) {
```

Comment:
what's the point?

Reply:
so similar to change candidate, if it's transferring to same owner, then don't need to modify the bucket, and can early exit